nomad/state/node_events.go

```diff
@@ -0,0 +1,175 @@
+package state
+
+import (
```

Comment:
this file and its corresponding tests are meant to exercise the transaction changes but will be changing greatly in future commits